src/Microsoft.AspNet.Razor/RazorError.cs

```diff
@@ -77,7 +77,8 @@ public bool Equals(RazorError other)
         {
             return other != null &&
                 string.Equals(other.Message, Message, StringComparison.Ordinal) &&
-                Location.Equals(other.Location);
+                Location.Equals(other.Location) &&
+                Length.Equals(other.Length);
```

Comment:
do not make this change.  `Length` can be changed and therefore cannot be part of the `RazorError`'s identity.